folium/vector_layers.py

```diff
@@ -66,27 +66,15 @@ def path_options(**kwargs):
     http://leafletjs.com/reference-1.2.0.html#path
 
     """
-    valid_options = (
```

Comment:
I get that it's cleaner to not have to define and maintain this list. But now if someone passes a faulty option it gets ignored silently, which is a problem because a user may not exactly know which options are valid. When no error is raised it's hard to find out. Can we add a check at the bottom of this function that warns when any kwargs have not been popped?

Reply:
I'm coming from a world where `kwargs` are a user responsibility and the valid options are well documented. The latter is not true in `folium` and not all user are like me :grimacing: So please take this with a grain of salt :wink: 

If we choose warning route we must revert the parsing of the `kwargs` in the plugins to avoid triggering warnings for valid plugin options. However, that may override the option with when `path_options` runs. (That was the case for antpath and the reason for this change.)

Maybe the best option is to create a `kwarg` check function that runs after all the options are parsed and run it in the plugin. What do you think? Not sure if that is worth it though...

PS: the reason for removing the default check was to accommodate the extra `kwargs` that are only valid for the antpath and the different defaults from polyline while still relying `path_options`.

Reply:
Alright I see it's not that easy. We'll leave it as it is now and maybe work on this in a separate PR.